src/Microsoft.ML.Data/Data/Conversion.cs

```diff
@@ -1285,7 +1298,12 @@ public bool TryParse(ref TX src, out I8 dst)
         {
             dst = default;
             TryParseSigned(I8.MaxValue, ref src, out long? res);
-            Contracts.Check(res.HasValue, "Value could not be parsed from text to long.");
```

Comment:
>Contracts. [](start = 12, length = 10)

should it warn, though?

Reply:
The returning bool being false is the warning. If something however *wishes* to warn upon seeing this, that is its business.